components/json/rs/src/lib.rs

```diff
@@ -0,0 +1,128 @@
+#[cxx::bridge(namespace = json)]
+mod ffi {
+    extern "Rust" {
+      fn convert_uint64_to_unicode_string(path: &str, json: &str) -> String;
+      fn convert_int64_to_unicode_string(path: &str, json: &str) -> String;
+      fn convert_unicode_string_to_uint64(path: &str, json: &str) -> String;
+      fn convert_unicode_string_to_int64(path: &str, json: &str) -> String;
+    }
+}
+
+// Converts uint64 value to string and serializes json with new values.
+// A path is a Unicode string with the reference tokens separated by /. Inside tokens / is replaced by ~1 and ~ is replaced by ~0.
+// Examples: convert_uint64_to_unicode_string("/a/b", json) for { a : { b : 1 }}
+//           convert_uint64_to_unicode_string("/a/0", json) for { a : [ 1 ]}
+pub fn convert_uint64_to_unicode_string(path: &str, json: &str) -> String {
+  let mut unwrapped_value: serde_json::Value = serde_json::from_str(&json).unwrap_or_else(|e| {
+    eprintln!("Failed to parse JSON: {}", e);
+    return serde_json::Value::Null;
+  });
+  if unwrapped_value.is_null() {
+    return String::new();
+  }
```

Comment:
Is this check redundant? I think the subsequent `pointer_mut` method will [return None on a null value](https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=c1d71d9f060dbe538eab6b6448898c16).

Reply:
In case anyone else is as confused as I was by `pointer` and `is_null` in the serde_json api, remember that Rust references never point to invalid memory, so this is *not* checking if the `&Value` or `&mut Value` type points to address zero. Instead it's checking if the *valid json value object* the reference points to represents the `null` json value, as in `{ "a": 123, "b": null }`.

Since it's just another enum variant, calling `is_null()` before reading the value isn't important the way it would be in C/C++ api returning a pointer.

Reply:
Discussed in slack, no additional changes required